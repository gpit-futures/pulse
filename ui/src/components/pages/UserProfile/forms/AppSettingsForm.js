import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateAppSettingsForm } from '../forms.validation';
import { valuesSettingsForm, valuesSettingsFormLabels } from '../forms.config';
import { setTheme } from '../../../../ducks/set-theme.duck';

@reduxForm({
  form: 'appSettingsFormSelector',
  validate: validateAppSettingsForm,
})
export default class AppSettingsForm extends PureComponent {
  static propTypes = {
    dispatch: PropTypes.func,
    initialize: PropTypes.func,
  };

  componentDidMount() {
    const { patientsInfo, initialize } = this.props;
    initialize(this.defaultValuesForm(patientsInfo));
  }

  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesSettingsForm.APP_TITLE]: value.title,
      [valuesSettingsForm.LOGO_PATH]: value.logoB64,
      [valuesSettingsForm.SELECT_THEME]: value.themeColor,
      [valuesSettingsForm.BROWSER_TITLE]: value.browserTitle,
    };

    return defaultFormValues;
  }

 setThemeHook = dispatch => (theme) => {
   dispatch(setTheme(theme));
   return theme;
 };

 render() {
   const { dispatch } = this.props;

   return (
     <div className="panel-body-inner">
       <form name="appSettingsForm" className="form">
         <div className="form-group-wrapper">
           <Row>
             <Col xs={12} md={6}>
               <Row>
                 <Col md={11}>
                   <Field
                     label={valuesSettingsFormLabels.APP_TITLE}
                     name={valuesSettingsForm.APP_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                   <Field
                     label={valuesSettingsFormLabels.LOGO_PATH}
                     name={valuesSettingsForm.LOGO_PATH}
                     component={FileInput}
                     id={valuesSettingsForm.LOGO_PATH}
                   />
                   <Field
                     label={valuesSettingsFormLabels.SELECT_THEME}
                     name={valuesSettingsForm.SELECT_THEME}
                     component={Select}
                     options={optionsForThemesField}
                     //here we're using normalize hook for dispatching action to set app theme
                     normalize={this.setThemeHook(dispatch)}
                   />
                   <Field
                     label={valuesSettingsFormLabels.BROWSER_TITLE}
                     name={valuesSettingsForm.BROWSER_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                 </Col>
               </Row>
             </Col>
           </Row>
         </div>
       </form>
     </div>)
 }
}
