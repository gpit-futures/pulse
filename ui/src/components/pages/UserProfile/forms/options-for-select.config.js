const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

const optionsForCountryField = [{
  value: 'usa',
  title: 'USA',
}, {
  value: 'country1',
  title: 'Country 1',
}, {
  value: 'country2',
  title: 'Country 2',
}, {
  value: 'country3',
  title: 'Country 3',
}, {
  value: 'country4',
  title: 'Country 4',
}, {
  value: 'country5',
  title: 'Country 5',
}];

export { optionsForThemesField, optionsForGenderField, optionsForCountryField }
