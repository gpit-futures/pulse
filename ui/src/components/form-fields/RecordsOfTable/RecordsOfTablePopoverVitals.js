import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-bootstrap';
import _ from 'lodash/fp';

import { valuesAddons, valuesLabels, valuesNames } from '../../pages/Vitals/forms.config';
import { serviceVitalsSigns } from '../../pages/Vitals/viltals-helpers.utils';
import Switch from '../Switch';

export default class RecordsOfTablePopoverVitals extends PureComponent {
  static propTypes = {
    detail: PropTypes.object,
  };
  state = {
    vitalStatuses: {
      [valuesNames.RESPIRATION_RATE]: {},
      [valuesNames.OXYGEN_SATURATION]: {},
      [valuesNames.HEART_RATE]: {},
      [valuesNames.SYSTOLIC_BP]: {},
      [valuesNames.TEMPERATURE]: {},
      [valuesNames.NEWS_SCORE]: {},
    },
  };

  componentWillMount() {
    const { detail } = this.props;
    this.setVitalStatuses(detail);
  }

  componentWillReceiveProps(nextProps) {
    if (!_.isEmpty(nextProps.detail)) {
      this.setVitalStatuses(nextProps.detail);
    }
  }

  getHighlighterClass = (vitalName) => {
    const { vitalStatuses } = this.state;
    return serviceVitalsSigns.getHighlighterClass(vitalStatuses[vitalName]);
  };

  setVitalStatuses = (vital) => {
    if (!_.isEmpty(vital)) {
      const vitalDetail = serviceVitalsSigns.convertVitalCharacteristics(vital);
      const vitalStatuses = serviceVitalsSigns.setVitalStatuses(vitalDetail);

      vitalStatuses.newsScore = serviceVitalsSigns.getStatusOnValue(vitalDetail.newsScore, 'newsScore');

      this.setState({ vitalStatuses });
    }
  };

  render() {
    let { detail } = this.props;
    const { vitalStatuses } = this.state;
    detail = detail || {};

    return (
      <div className="form-group-wrapper">
        <div className="row-expand">
          <div className="col-expand-left">
            <Row>
              <Col xs={12} md={6}>
                <div className="vitals-group highlighter-wrapper">
                  <span className={`${this.getHighlighterClass(valuesNames.RESPIRATION_RATE)}`} />
                  <label className="vitals-label">{valuesLabels.RESPIRATION_RATE}</label>
                  <div className={`input-group vitals-holder popover-wrap ${vitalStatuses[valuesNames.RESPIRATION_RATE].type}`} >
                    <div className="form-control input-sm">{detail[valuesNames.RESPIRATION_RATE]}</div>
                    <span className="vitals-addon">{valuesAddons.RESPIRATION_RATE}</span>
                  </div>
                </div>
              </Col>
              <Col xs={12} md={6}>
                <div className="vitals-group highlighter-wrapper">
                  <span className={`${this.getHighlighterClass(valuesNames.OXYGEN_SATURATION)}`} />
                  <label className="vitals-label">{valuesLabels.OXYGEN_SATURATION}</label>
                  <div className={`input-group vitals-holder popover-wrap ${vitalStatuses[valuesNames.OXYGEN_SATURATION].type}`} >
                    <div className="form-control input-sm">{detail[valuesNames.OXYGEN_SATURATION]}</div>
                    <span className="vitals-addon">{valuesAddons.OXYGEN_SATURATION}</span>
                  </div>
                </div>
              </Col>
            </Row>
            <Row>
              <Col xs={12} md={6}>
                <div className="vitals-group highlighter-wrapper">
                  <span className={`${this.getHighlighterClass(valuesNames.OXYGEN_SUPPLEMENTAL)}`} />
                  <label className="vitals-label">{valuesLabels.OXYGEN_SUPPLEMENTAL}</label>
                  <div className="input-holder">
                    <Switch
                      type="checkbox"
                      name={valuesNames.OXYGEN_SUPPLEMENTAL}
                      disabled
                      value={detail[valuesNames.OXYGEN_SUPPLEMENTAL]}
                    />
                  </div>
                </div>
              </Col>
              <Col xs={12} md={6}>
                <div className="vitals-group highlighter-wrapper">
                  <span className={`${this.getHighlighterClass(valuesNames.HEART_RATE)}`} />
                  <label className="vitals-label">{valuesLabels.HEART_RATE}</label>
                  <div className={`input-group vitals-holder popover-wrap ${vitalStatuses[valuesNames.HEART_RATE].type}`} >
                    <div className="form-control input-sm">{detail[valuesNames.HEART_RATE]}</div>
                    <span className="vitals-addon">{valuesAddons.HEART_RATE}</span>
                  </div>
                </div>
              </Col>
            </Row>
          </div>
          <div className="col-expand-right">
            <Row>
              <Col xs={12} md={6}>
                <div className="vitals-group highlighter-wrapper">
                  <span className={`${this.getHighlighterClass(valuesNames.SYSTOLIC_BP)}`} />
                  <label className="vitals-label">{valuesLabels.SYSTOLIC_BP}</label>
                  <div className={`input-group vitals-holder popover-wrap ${vitalStatuses[valuesNames.SYSTOLIC_BP].type}`} >
                    <div className="form-control input-sm">{detail[valuesNames.SYSTOLIC_BP]}</div>
                    <span className="vitals-addon">{valuesAddons.SYSTOLIC_BP}</span>
                  </div>
                </div>
              </Col>
              <Col xs={12} md={6}>
                <div className="vitals-group highlighter-wrapper">
                  <span className="highlighter-not-vital" />
                  <label className="vitals-label">{valuesLabels.DIASTOLIC_BP}</label>
                  <div className="input-group vitals-holder">
                    <div className="form-control input-sm">{detail[valuesNames.DIASTOLIC_BP]}</div>
                    <span className="vitals-addon">{valuesAddons.DIASTOLIC_BP}</span>
                  </div>
                </div>
              </Col>
            </Row>
            <Row>
              <Col xs={12} md={6}>
                <div className="vitals-group highlighter-wrapper">
                  <span className={`${this.getHighlighterClass(valuesNames.LEVEL_OF_CONSCIOUSNESS)}`} />
                  <label htmlFor={valuesNames.LEVEL_OF_CONSCIOUSNESS} className="vitals-label">{valuesLabels.LEVEL_OF_CONSCIOUSNESS}</label>
                  <div className="input-holder">
                    <div className="switch-group">
                      <Switch
                        type="radio"
                        name={valuesNames.LEVEL_OF_CONSCIOUSNESS}
                        disabled
                        value="Alert"
                        transitionValue={detail[valuesNames.LEVEL_OF_CONSCIOUSNESS]}
                        text="A"
                        id="levelOfConsciousnessA"
                      />
                      <Switch
                        type="radio"
                        name={valuesNames.LEVEL_OF_CONSCIOUSNESS}
                        disabled
                        value="Voice"
                        transitionValue={detail[valuesNames.LEVEL_OF_CONSCIOUSNESS]}
                        text="V"
                        id="levelOfConsciousnessV"
                      />
                      <Switch
                        type="radio"
                        name={valuesNames.LEVEL_OF_CONSCIOUSNESS}
                        disabled
                        value="Pain"
                        transitionValue={detail[valuesNames.LEVEL_OF_CONSCIOUSNESS]}
                        text="P"
                        id="levelOfConsciousnessP"
                      />
                      <Switch
                        type="radio"
                        name={valuesNames.LEVEL_OF_CONSCIOUSNESS}
                        disabled
                        value="Unresponsive"
                        transitionValue={detail[valuesNames.LEVEL_OF_CONSCIOUSNESS]}
                        text="U"
                        id="levelOfConsciousnessU"
                      />
                    </div>
                  </div>
                </div>
              </Col>
              <Col xs={12} md={6}>
                <div className="vitals-group highlighter-wrapper">
                  <span className={`${this.getHighlighterClass(valuesNames.TEMPERATURE)}`} />
                  <label className="vitals-label">{valuesLabels.TEMPERATURE}</label>
                  <div className={`input-group vitals-holder popover-wrap ${vitalStatuses[valuesNames.TEMPERATURE].type}`} >
                    <div className="form-control input-sm">{detail[valuesNames.TEMPERATURE]}</div>
                    <span className="vitals-addon">{valuesAddons.TEMPERATURE}</span>
                  </div>
                </div>
              </Col>
            </Row>
          </div>
        </div>
        <div className="row-expand">
          <div className="col-expand-left">
            <div className="vitals-group highlighter-wrapper">
              <span className={`${this.getHighlighterClass(valuesNames.NEWS_SCORE)}`} />
              <label className="vitals-label">{valuesLabels.NEWS_SCORE}</label>
              <div className={`input-holder vitals-holder ${vitalStatuses[valuesNames.NEWS_SCORE].type}`}>
                <div className="form-control input-sm">{detail[valuesNames.NEWS_SCORE]}</div>
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}
