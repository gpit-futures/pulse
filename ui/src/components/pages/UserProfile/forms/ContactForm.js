import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from '../forms.validation';
import { valuesContactForm, valuesContactFormLabels } from '../forms.config';
import { defaultContactFormValues } from './default-values.config';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultContactFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="contactForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesContactFormLabels.ADDRESS}
                      name={valuesContactForm.ADDRESS}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.CITY}
                      name={valuesContactForm.CITY}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.STATE}
                      name={valuesContactForm.STATE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.POSTAL_CODE}
                      name={valuesContactForm.POSTAL_CODE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.SELECT_COUNTRY}
                      name={valuesContactForm.SELECT_COUNTRY}
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesContactFormLabels.PHONE}
                      name={valuesContactForm.PHONE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.EMAIL}
                      name={valuesContactForm.EMAIL}
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
