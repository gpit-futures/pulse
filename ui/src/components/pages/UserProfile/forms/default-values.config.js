import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from '../forms.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
  [valuesPersonalForm.DATE_OF_BIRTH]: 0,
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
