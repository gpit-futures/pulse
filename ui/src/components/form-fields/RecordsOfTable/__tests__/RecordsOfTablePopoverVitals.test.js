import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import moment from 'moment';

import RecordsOfTablePopoverVitals from '../../../form-fields/RecordsOfTable/RecordsOfTablePopoverVitals';
import { valuesLabels, valuesNames, valuesAddons } from '../../../pages/Vitals/forms.config';

Enzyme.configure({ adapter: new Adapter() });

const DATE_FORMAT = 'DD-MMM-YYYY';

const DATE_TO_USE = new Date('2018');
const DATE_TO_USE_TIME = DATE_TO_USE.getTime();
global.Date = jest.fn(() => DATE_TO_USE);
Date.now = jest.fn(() => DATE_TO_USE_TIME);

const testProps = {
  detail: {
    [valuesNames.RESPIRATION_RATE]: '12.0',
    [valuesNames.OXYGEN_SUPPLEMENTAL]: 'false',
    [valuesNames.HEART_RATE]: '45.0',
    [valuesNames.TEMPERATURE]: '37.0',
    [valuesNames.LEVEL_OF_CONSCIOUSNESS]: 'Voice',
    [valuesNames.SYSTOLIC_BP]: '112.0',
    [valuesNames.DIASTOLIC_BP]: '64.0',
    [valuesNames.OXYGEN_SATURATION]: '97.0',
    [valuesNames.NEWS_SCORE]: '4',
    [valuesNames.AUTHOR]: 'Dr Tony Shannon',
    [valuesNames.DATE_CREATED]: 1516367377000,
    [valuesNames.SOURCE]: 'ethercis',
    [valuesNames.SOURCE_ID]: '27ee5e25-4c32-46d2-b45a-f74149d72030',
  },
};

const DATE_CREATED = moment(testProps.detail[valuesNames.DATE_CREATED]).format(DATE_FORMAT);

describe('Component <RecordsOfTablePopoverVitals />', () => {
  it('should renders with props correctly', () => {
    const component = shallow(
      <RecordsOfTablePopoverVitals
        detail={testProps.detail}
      />);

    expect(component.find('label')).toHaveLength(9);
    expect(component.find('.form-control')).toHaveLength(7);
    expect(component.find('.vitals-addon')).toHaveLength(6);
    expect(component.find('Switch')).toHaveLength(5);

    expect(component.find('label').at(0).text()).toEqual(valuesLabels.RESPIRATION_RATE);
    expect(component.find('label').at(1).text()).toEqual(valuesLabels.OXYGEN_SATURATION);
    expect(component.find('label').at(2).text()).toEqual(valuesLabels.OXYGEN_SUPPLEMENTAL);
    expect(component.find('label').at(3).text()).toEqual(valuesLabels.HEART_RATE);
    expect(component.find('label').at(4).text()).toEqual(valuesLabels.SYSTOLIC_BP);
    expect(component.find('label').at(5).text()).toEqual(valuesLabels.DIASTOLIC_BP);
    expect(component.find('label').at(6).text()).toEqual(valuesLabels.LEVEL_OF_CONSCIOUSNESS);
    expect(component.find('label').at(7).text()).toEqual(valuesLabels.TEMPERATURE);
    expect(component.find('label').at(8).text()).toEqual(valuesLabels.NEWS_SCORE);

    expect(component.find('.form-control').at(0).text()).toEqual(testProps.detail[valuesNames.RESPIRATION_RATE]);
    expect(component.find('.form-control').at(1).text()).toEqual(testProps.detail[valuesNames.OXYGEN_SATURATION]);
    expect(component.find('.form-control').at(2).text()).toEqual(testProps.detail[valuesNames.HEART_RATE]);
    expect(component.find('.form-control').at(3).text()).toEqual(testProps.detail[valuesNames.SYSTOLIC_BP]);
    expect(component.find('.form-control').at(4).text()).toEqual(testProps.detail[valuesNames.DIASTOLIC_BP]);
    expect(component.find('.form-control').at(5).text()).toEqual(testProps.detail[valuesNames.TEMPERATURE]);
    expect(component.find('.form-control').at(6).text()).toEqual(testProps.detail[valuesNames.NEWS_SCORE]);

    expect(component.find('.vitals-addon').at(0).text()).toEqual(valuesAddons.RESPIRATION_RATE);
    expect(component.find('.vitals-addon').at(1).text()).toEqual(valuesAddons.OXYGEN_SATURATION);
    expect(component.find('.vitals-addon').at(2).text()).toEqual(valuesAddons.HEART_RATE);
    expect(component.find('.vitals-addon').at(3).text()).toEqual(valuesAddons.SYSTOLIC_BP);
    expect(component.find('.vitals-addon').at(4).text()).toEqual(valuesAddons.DIASTOLIC_BP);
    expect(component.find('.vitals-addon').at(5).text()).toEqual(valuesAddons.TEMPERATURE);

    expect(component.find('Switch').at(0).props().value).toEqual(testProps.detail[valuesNames.OXYGEN_SUPPLEMENTAL]);
    expect(component.find('Switch').at(1).props().value).toEqual('Alert');
    expect(component.find('Switch').at(2).props().value).toEqual('Voice');
    expect(component.find('Switch').at(3).props().value).toEqual('Pain');
    expect(component.find('Switch').at(4).props().value).toEqual('Unresponsive');

    expect(component).toMatchSnapshot();

    // Testing lifecycle method
    component.setProps({ test: 'test' });
    component.setProps({ detail: {} });
  });

  it('should renders with props correctly with empty detail', () => {
    const component = shallow(
      <RecordsOfTablePopoverVitals />);

    expect(component).toMatchSnapshot();
  });
});

