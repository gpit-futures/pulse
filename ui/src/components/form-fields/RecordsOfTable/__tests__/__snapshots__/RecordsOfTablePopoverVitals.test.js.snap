// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <RecordsOfTablePopoverVitals /> should renders with props correctly 1`] = `
<div
  className="form-group-wrapper"
>
  <div
    className="row-expand"
  >
    <div
      className="col-expand-left"
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital"
            />
            <label
              className="vitals-label"
            >
              Respiration Rate
            </label>
            <div
              className="input-group vitals-holder popover-wrap undefined"
            >
              <div
                className="form-control input-sm"
              >
                12.0
              </div>
              <span
                className="vitals-addon"
              >
                resps/min
              </span>
            </div>
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital"
            />
            <label
              className="vitals-label"
            >
              Oxygen Saturation
            </label>
            <div
              className="input-group vitals-holder popover-wrap undefined"
            >
              <div
                className="form-control input-sm"
              >
                97.0
              </div>
              <span
                className="vitals-addon"
              >
                %
              </span>
            </div>
          </div>
        </Col>
      </Row>
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital"
            />
            <label
              className="vitals-label"
            >
              Any Supplemental Oxygen
            </label>
            <div
              className="input-holder"
            >
              <Switch
                disabled={true}
                name="oxygenSupplemental"
                type="checkbox"
                value="false"
              />
            </div>
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-success"
            />
            <label
              className="vitals-label"
            >
              Heart Rate
            </label>
            <div
              className="input-group vitals-holder popover-wrap success"
            >
              <div
                className="form-control input-sm"
              >
                45.0
              </div>
              <span
                className="vitals-addon"
              >
                bpm
              </span>
            </div>
          </div>
        </Col>
      </Row>
    </div>
    <div
      className="col-expand-right"
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital"
            />
            <label
              className="vitals-label"
            >
              Systolic BP
            </label>
            <div
              className="input-group vitals-holder popover-wrap undefined"
            >
              <div
                className="form-control input-sm"
              >
                112.0
              </div>
              <span
                className="vitals-addon"
              >
                mmHg
              </span>
            </div>
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital"
            />
            <label
              className="vitals-label"
            >
              Diastolic BP
            </label>
            <div
              className="input-group vitals-holder"
            >
              <div
                className="form-control input-sm"
              >
                64.0
              </div>
              <span
                className="vitals-addon"
              >
                mmHg
              </span>
            </div>
          </div>
        </Col>
      </Row>
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-danger"
            />
            <label
              className="vitals-label"
              htmlFor="levelOfConsciousness"
            >
              Level of Consciousness
            </label>
            <div
              className="input-holder"
            >
              <div
                className="switch-group"
              >
                <Switch
                  disabled={true}
                  id="levelOfConsciousnessA"
                  name="levelOfConsciousness"
                  text="A"
                  transitionValue="Voice"
                  type="radio"
                  value="Alert"
                />
                <Switch
                  disabled={true}
                  id="levelOfConsciousnessV"
                  name="levelOfConsciousness"
                  text="V"
                  transitionValue="Voice"
                  type="radio"
                  value="Voice"
                />
                <Switch
                  disabled={true}
                  id="levelOfConsciousnessP"
                  name="levelOfConsciousness"
                  text="P"
                  transitionValue="Voice"
                  type="radio"
                  value="Pain"
                />
                <Switch
                  disabled={true}
                  id="levelOfConsciousnessU"
                  name="levelOfConsciousness"
                  text="U"
                  transitionValue="Voice"
                  type="radio"
                  value="Unresponsive"
                />
              </div>
            </div>
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital"
            />
            <label
              className="vitals-label"
            >
              Temperature
            </label>
            <div
              className="input-group vitals-holder popover-wrap undefined"
            >
              <div
                className="form-control input-sm"
              >
                37.0
              </div>
              <span
                className="vitals-addon"
              >
                c
              </span>
            </div>
          </div>
        </Col>
      </Row>
    </div>
  </div>
  <div
    className="row-expand"
  >
    <div
      className="col-expand-left"
    >
      <div
        className="vitals-group highlighter-wrapper"
      >
        <span
          className="highlighter-success"
        />
        <label
          className="vitals-label"
        >
          NEWS Score
        </label>
        <div
          className="input-holder vitals-holder success"
        >
          <div
            className="form-control input-sm"
          >
            4
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <RecordsOfTablePopoverVitals /> should renders with props correctly with empty detail 1`] = `
<div
  className="form-group-wrapper"
>
  <div
    className="row-expand"
  >
    <div
      className="col-expand-left"
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital"
            />
            <label
              className="vitals-label"
            >
              Respiration Rate
            </label>
            <div
              className="input-group vitals-holder popover-wrap undefined"
            >
              <div
                className="form-control input-sm"
              />
              <span
                className="vitals-addon"
              >
                resps/min
              </span>
            </div>
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital"
            />
            <label
              className="vitals-label"
            >
              Oxygen Saturation
            </label>
            <div
              className="input-group vitals-holder popover-wrap undefined"
            >
              <div
                className="form-control input-sm"
              />
              <span
                className="vitals-addon"
              >
                %
              </span>
            </div>
          </div>
        </Col>
      </Row>
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital;"
            />
            <label
              className="vitals-label"
            >
              Any Supplemental Oxygen
            </label>
            <div
              className="input-holder"
            >
              <Switch
                disabled={true}
                name="oxygenSupplemental"
                type="checkbox"
              />
            </div>
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital"
            />
            <label
              className="vitals-label"
            >
              Heart Rate
            </label>
            <div
              className="input-group vitals-holder popover-wrap undefined"
            >
              <div
                className="form-control input-sm"
              />
              <span
                className="vitals-addon"
              >
                bpm
              </span>
            </div>
          </div>
        </Col>
      </Row>
    </div>
    <div
      className="col-expand-right"
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital"
            />
            <label
              className="vitals-label"
            >
              Systolic BP
            </label>
            <div
              className="input-group vitals-holder popover-wrap undefined"
            >
              <div
                className="form-control input-sm"
              />
              <span
                className="vitals-addon"
              >
                mmHg
              </span>
            </div>
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital"
            />
            <label
              className="vitals-label"
            >
              Diastolic BP
            </label>
            <div
              className="input-group vitals-holder"
            >
              <div
                className="form-control input-sm"
              />
              <span
                className="vitals-addon"
              >
                mmHg
              </span>
            </div>
          </div>
        </Col>
      </Row>
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital;"
            />
            <label
              className="vitals-label"
              htmlFor="levelOfConsciousness"
            >
              Level of Consciousness
            </label>
            <div
              className="input-holder"
            >
              <div
                className="switch-group"
              >
                <Switch
                  disabled={true}
                  id="levelOfConsciousnessA"
                  name="levelOfConsciousness"
                  text="A"
                  type="radio"
                  value="Alert"
                />
                <Switch
                  disabled={true}
                  id="levelOfConsciousnessV"
                  name="levelOfConsciousness"
                  text="V"
                  type="radio"
                  value="Voice"
                />
                <Switch
                  disabled={true}
                  id="levelOfConsciousnessP"
                  name="levelOfConsciousness"
                  text="P"
                  type="radio"
                  value="Pain"
                />
                <Switch
                  disabled={true}
                  id="levelOfConsciousnessU"
                  name="levelOfConsciousness"
                  text="U"
                  type="radio"
                  value="Unresponsive"
                />
              </div>
            </div>
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
          xs={12}
        >
          <div
            className="vitals-group highlighter-wrapper"
          >
            <span
              className="highlighter-not-vital"
            />
            <label
              className="vitals-label"
            >
              Temperature
            </label>
            <div
              className="input-group vitals-holder popover-wrap undefined"
            >
              <div
                className="form-control input-sm"
              />
              <span
                className="vitals-addon"
              >
                c
              </span>
            </div>
          </div>
        </Col>
      </Row>
    </div>
  </div>
  <div
    className="row-expand"
  >
    <div
      className="col-expand-left"
    >
      <div
        className="vitals-group highlighter-wrapper"
      >
        <span
          className="highlighter-not-vital"
        />
        <label
          className="vitals-label"
        >
          NEWS Score
        </label>
        <div
          className="input-holder vitals-holder undefined"
        >
          <div
            className="form-control input-sm"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
